src/View/Helper/DateHelper.php

```diff
@@ -137,13 +137,13 @@ public function ago($date, $alone=true)
                 );
             }
         } elseif ($diff->days > 0) {
-            return format(__n('yesterday', '{n}&nbsp;days ago', $diff->days), array('n' => $diff->days));
+            return format(__n('yesterday', "{n}\u{00A0}days ago", $diff->days), array('n' => $diff->days));
         } elseif ($diff->h > 0) {
-            return format(__n('an hour ago', '{n}&nbsp;hours ago', $diff->h), array('n' => $diff->h));
+            return format(__n('an hour ago', "{n}\u{00A0}hours ago", $diff->h), array('n' => $diff->h));
         } else {
             // we stop at minute accuracy
             $minutes = max($diff->i, 1);
-            return format(__n('a minute ago', '{n}&nbsp;minutes ago', $minutes), array('n' => $minutes));
```

Comment:
Thanks, I see! The existing code base mostly assumes translations can contain HTML, as it is sometimes necessary to include formatting tags such as `<a>`, `<em>` or `<strong>`.

So I am concerned that this change would invalidate a lot of translations that won’t be added back before long in many languages since we don’t have a dedicated team of UI translators. For example, let’s say a contributor translated the UI up to 100% in Korean (just an example right) and then does not actively maintain the translation, so every time some new translatable strings are added or replaced, the translation coverage decreases to 99%, 98%, 95% etc. These date strings are quite visible because they are used in a lot of places, so updating them will have a substantial impact on Tatoeba’s outlook in many UI languages for some time.

I am also concerned about the discrepancy it creates. For example some translators may nevertheless add some `&nbsp;` in their translation of "{n} days ago" and see that it works it one place, such as the Wall, and conclude it’s fine, whereas it’s not displaying correctly in the audio block.

From a security standpoint it’s not very secure to allow HTML from translations, so what AngularJS is doing by default is actually the right thing to do. But this is not the way we deal with translations for now on Tatoeba, so I suggest using AngularJS’s `trustAsHtml()` instead to keep things consistent between AngularJS code and PHP code.

Reply:
Ah, okay. I wasn't aware it would break translations like that. I'm not sure how I feel about blindly trusting the translations with `trustAsHtml()`. I feel like it's overkill since the only issue is a single html entity. Since this is an isolated incident, what if the `&nbsp;` entity is instead replaced with the Unicode NBSP just before Angular renders them. Like this.

### audio-details.ctrl.js
```js
function init(audios, audioLicenses) {
    vm.audios = audios.map((audio) => {
        if (audio.created_ago) {
            audio.created_ago = audio.created_ago.replace("&nbsp;", "\u00A0")
        }
        if (audio.modified_ago) {
            audio.modified_ago = audio.modified_ago.replace("&nbsp;", "\u00A0")
        }
        return audio
    })
    vm.audioLicenses = audioLicenses;
}
```

This way we don't have to worry about the security implications of using `trustAsHtml()` for any translations.

Reply:
@jiru ^ would this be an acceptable solution?

Reply:
> This way we don't have to worry about the security implications of using trustAsHtml() for any translations.

Regardless of this PR, we already have to worry about the security implications of trusting translations as HTML in pretty much all the frontend code that is not AngularJS-ified. We do this by manually checking translated strings when committing `.po` files so it’s not a such big worry.

> would this be an acceptable solution?

Sorry but I don’t think this is a good way to approach the problem. I’d like to suggest 2 alternatives:
* leave the translatable string with `&nbsp;` and use `trustAsHtml()` (I can commit that specific change if you want to avoid responsibility for any security implication)
* change `&nbsp;` to `\u{00A0}` and I may be able to do some regexp magic on Transifex to keep all translations unaffected

Reply:
Ah okay. I understand. If it's an implicit assumption throughout the app, you're right it doesn't really make a difference for this PR. As long as there is awareness of it. I'll go ahead and revert the `DateHelper` to keep using `&nbsp;` and use `trustAsHtml()` as recommended.

Reply:
Okay, I've reverted to using the `&nbsp;`. I elected to use the `ng-bind-html` Angular attribute to allow the HTML rendering since it seemed more in line with the existing code than including the `$sce.trustAsHtml()`.